src/CoreFoundation/CFProxySupport.cs

```diff
@@ -602,19 +609,27 @@ public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (NSString prox
 
 			if (resultCallback == null)
 				throw new ArgumentNullException ("resultCallback");
-
-			if (clientContext == null)
-				throw new ArgumentNullException ("clientContext");
-
-			IntPtr source = CFNetworkExecuteProxyAutoConfigurationScript (proxyAutoConfigurationScript.Handle, targetURL.Handle, resultCallback, clientContext);
-			return (source == IntPtr.Zero) ? null : new CFRunLoopSource (source);
+			unsafe {
+				BlockLiteral block_handler = new BlockLiteral ();
+				BlockLiteral *block_ptr_handler = &block_handler;
+				block_handler.SetupBlockUnsafe (static_AutoConfigurationResult, resultCallback);
+		
+				try {
+					IntPtr clientPtr = Marshal.AllocHGlobal (Marshal.SizeOf (clientContext));
+					Marshal.StructureToPtr (clientContext, clientPtr, false);
+					IntPtr source = CFNetworkExecuteProxyAutoConfigurationScript (proxyAutoConfigurationScript.Handle, targetURL.Handle, (void*) block_ptr_handler, clientPtr);
```

Comment:
Do: `ref block_handler` instead of `(void*) block_ptr_handler` and you can remove the `block_ptr_handler` variable and the `unsafe` block.